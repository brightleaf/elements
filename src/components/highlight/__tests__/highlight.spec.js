import React from 'react'
import { render, cleanup } from '@testing-library/react'
import {
  toHaveClass,
  toHaveAttribute,
  toContainElement,
} from '@testing-library/jest-dom'
import { Highlight } from '../'

expect.extend({ toHaveClass, toHaveAttribute, toContainElement })
describe('The Highlight Component', () => {
  afterEach(cleanup)
  it('should render a div with the class highlight', () => {
    const { container } = render(<Highlight> </Highlight>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('highlight')
  })
  it('should render a div with the class highlight and the className passed in', () => {
    const { container } = render(<Highlight className="highlight-test" />)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('highlight')
    expect(container.firstChild).toHaveClass('highlight-test')
  })
  it('should render a p with the class highlight when as property is set to `p`', () => {
    const { container } = render(<Highlight as="p" />)
    expect(container).toMatchSnapshot()

    expect(container.firstChild).toHaveClass('highlight')
    const p = document.querySelector('p.highlight')
    expect(container).toContainElement(p)
  })
})
