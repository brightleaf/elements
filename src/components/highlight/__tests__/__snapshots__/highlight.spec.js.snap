// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Highlight Component should render a div with the class highlight 1`] = `
<div
  class="highlight"
>
   
</div>
`;

exports[`The Highlight Component should render a div with the class highlight and the className passed in 1`] = `
<div
  class="highlight highlight-test"
/>
`;

exports[`The Highlight Component should render a p with the class highlight when as property is set to \`p\` 1`] = `
<div>
  <p
    class="highlight"
  />
</div>
`;
